src/components/molecules/TableHeader/components/EditTableTitleModal/EditTableTitleModal.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useCallback, useState } from "react";
+import { Modal } from "react-bootstrap";
+
+import "./EditTableTitleModal.scss";
+
+export interface EditTableTitleModalProps {
+  isShown: boolean;
+  title: string;
+  error: string;
+  onHide: () => void;
+  onCancel: () => void;
+  onSave: (newTableTitle: string) => void;
```

Comment:
If we're going to have callbacks, they should almost always be optional. Otherwise they just end up getting passed noop functions.

Reply:
I agree with you. But this component is not a common/reusable component. I don't think this is ever going to be used outside of this scope (TableHeader). And in this specific case it should always have these callbacks.

Reply:
Mm, fair enough.

I guess see also my other comments semi-related to this RE: wondering if we should even be passing in such generic handles here if it's only single use. Rather than moving that functionality inside the component itself.

Reply:
It's been moved inside, good point: https://github.com/sparkletown/sparkle/pull/1377/commits/4b7532e6f02068126c212a6f0dab1714aaaad55a